addon/macros/t.ts

```diff
@@ -15,7 +15,7 @@ function partitionDynamicValuesAndStaticValues(options: Record<string, string |
     if (value instanceof Raw) {
       staticValues[key] = value.valueOf();
     } else {
-      dynamicValues[key] = value;
+      dynamicValues[key] = value!;
```

Comment:
Here we could check that `value` is not `undefined`, but again, that would change which keys would be added to `dynamicValues`.

We could update `dynamicValues` to allow `undefined` values

```ts
const dynamicValues = new EmptyObject() as Record<string, string | undefined>;
```

Reply:
let's exclude `undefined`